### PR TITLE
Remove some unnecessary allocations / copies on startup

### PR DESCRIPTION
**Pull request by elinor-fung**
Noticed some unnecessary allocations from when we parse the TPA paths, so I decided to do a pass to look for / remove some others.

On my Windows machine, this was minus ~3k native heap allocations for launching a blank framework-dependent console app.

### CODE REVIEW

File: src/native/corehost/deps_format.cpp [hunk 1]
```diff
@@ -96,20 +96,21 @@ void deps_json_t::reconcile_libraries_with_targets(
                 entry.asset = asset;
                 entry.asset.name = asset_name;
 
-                m_deps_entries[i].push_back(entry);
+                m_deps_entries[i].push_back(std::move(entry));
 
                 if (trace::is_enabled())
                 {
+                    const deps_entry_t& parsed = m_deps_entries[i].back();
```
Comment (VSadov) [inline src/native/corehost/deps_format.cpp:103]: Maybe if push_back/move happens after `if (trace::is_enabled())` block, we would not need `parsed` ?

File: src/native/corehost/hostpolicy/deps_resolver.cpp [hunk 2]
```diff
@@ -90,21 +90,22 @@ namespace
...
-                resolved_asset.resolved_path.c_str(),
-                resolved_asset.asset.assembly_version.as_str().c_str(),
-                resolved_asset.asset.file_version.as_str().c_str());
+                resolved_path.c_str(),
+                asset.assembly_version.as_str().c_str(),
+                asset.file_version.as_str().c_str());
         }
 
-        items->emplace(resolved_asset.asset.name, resolved_asset);
+        items->emplace(asset.name, deps_resolved_asset_t(asset, resolved_path));
```
Comment (VSadov) [inline src/native/corehost/hostpolicy/deps_resolver.cpp:108]: Is this the place where copies were happening? 
(Making the argument an rvalue bypasses copying - Is that the idea of the change?)

Comment (elinor-fung) [reply src/native/corehost/hostpolicy/deps_resolver.cpp:108]: Yeah. Also figured we should avoid creating the `deps_resolved_asset_t` if we weren't going to need it (if the asset name is already in the map).

**Review by VSadov — APPROVED** [review]
LGTM. Thanks!!

**Review by vitek-karas — APPROVED** [review]

**Pull request merged**